File: frontend-design/scripts/generate_identity_brief.py
#!/usr/bin/env python3
"""Generate identity package / CIP deliverable guidance.

Usage:
  python scripts/generate_identity_brief.py "consulting startup business card signage"
"""

from __future__ import annotations

import argparse
from _catalog import rank_csv, pick, md_bullet


def main() -> int:
    parser = argparse.ArgumentParser(description="Generate identity package brief")
    parser.add_argument("query", nargs="+")
    parser.add_argument("-n", "--limit", type=int, default=4)
    args = parser.parse_args()
    query = " ".join(args.query)
    deliverables = rank_csv("identity-deliverables.csv", query, args.limit)
    industries = rank_csv("identity-industries.csv", query, 2)
    styles = rank_csv("identity-styles.csv", query, 2)
    mockups = rank_csv("identity-mockup-contexts.csv", query, 2)
    print(f"# Identity Package Brief: {query}\n")
    if industries:
        print("## Industry Direction\n")
        for row in industries:
            print(f"### {pick(row, 'Industry')}")
            print(md_bullet("CIP style", pick(row, "CIP Style")))
            print(md_bullet("Primary colors", pick(row, "Primary Colors")))
            print(md_bullet("Typography", pick(row, "Typography")))
            print(md_bullet("Key deliverables", pick(row, "Key Deliverables")))
            print("")
    if styles:
        print("## Identity Styles\n")
        for row in styles:
            print(f"### {pick(row, 'Style Name')}")
            print(md_bullet("Description", pick(row, "Description")))
            print(md_bullet("Colors", pick(row, "Primary Colors")))
            print(md_bullet("Typography", pick(row, "Typography")))
            print(md_bullet("Materials", pick(row, "Materials")))
            print("")
    if deliverables:
        print("## Deliverables\n")
        for row in deliverables:
            print(f"### {pick(row, 'Deliverable')}")
            print(md_bullet("Category", pick(row, "Category")))
            print(md_bullet("Description", pick(row, "Description")))
            print(md_bullet("Dimensions", pick(row, "Dimensions")))
            print(md_bullet("Format", pick(row, "File Format")))
            print(md_bullet("Logo placement", pick(row, "Logo Placement")))
            print("")
    if mockups:
        print("## Mockup Contexts\n")
        for row in mockups:
            print(f"### {pick(row, 'Context Name')}")
            print(md_bullet("Scene", pick(row, "Scene Description")))
            print(md_bullet("Lighting", pick(row, "Lighting")))
            print(md_bullet("Environment", pick(row, "Environment")))
            print(md_bullet("Props", pick(row, "Props")))
            print("")
    print("## Handoff Rules\n")
    print("- Include logo variants, palette, typography, and usage rules.")
    print("- Specify dimensions, file formats, material/finish assumptions, and production constraints.")
    print("- Separate digital assets, print assets, source files, and exports.")
    print("- Keep naming and versioning stable for approvals.")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: frontend-design/scripts/generate_identity_brief.mjs
#!/usr/bin/env node
/** Generate identity package and CIP deliverable guidance. */

import { pathToFileURL } from "node:url";
import { mdBullet, pick, rankCsv } from "./_catalog.mjs";

function parseArgs(args) {
  let limit = 4;
  const query = [];
  for (let index = 0; index < args.length; index += 1) {
    if (args[index] === "-n" || args[index] === "--limit") {
      limit = Number.parseInt(args[index + 1], 10);
      index += 1;
    } else query.push(args[index]);
  }
  if (query.length === 0 || !Number.isInteger(limit)) throw new Error("Usage: generate_identity_brief.mjs <query> [-n <limit>]");
  return { query: query.join(" "), limit };
}

export function main(args = process.argv.slice(2)) {
  let options;
  try {
    options = parseArgs(args);
  } catch (error) {
    console.error(error.message);
    return 2;
  }
  const deliverables = rankCsv("identity-deliverables.csv", options.query, options.limit);
  const industries = rankCsv("identity-industries.csv", options.query, 2);
  const styles = rankCsv("identity-styles.csv", options.query, 2);
  const mockups = rankCsv("identity-mockup-contexts.csv", options.query, 2);
  const lines = [`# Identity Package Brief: ${options.query}`, ""];
  if (industries.length > 0) {
    lines.push("## Industry Direction", "");
    for (const row of industries) lines.push(`### ${pick(row, "Industry")}`, mdBullet("CIP style", pick(row, "CIP Style")), mdBullet("Primary colors", pick(row, "Primary Colors")), mdBullet("Typography", pick(row, "Typography")), mdBullet("Key deliverables", pick(row, "Key Deliverables")), "");
  }
  if (styles.length > 0) {
    lines.push("## Identity Styles", "");
    for (const row of styles) lines.push(`### ${pick(row, "Style Name")}`, mdBullet("Description", pick(row, "Description")), mdBullet("Colors", pick(row, "Primary Colors")), mdBullet("Typography", pick(row, "Typography")), mdBullet("Materials", pick(row, "Materials")), "");
  }
  if (deliverables.length > 0) {
    lines.push("## Deliverables", "");
    for (const row of deliverables) lines.push(`### ${pick(row, "Deliverable")}`, mdBullet("Category", pick(row, "Category")), mdBullet("Description", pick(row, "Description")), mdBullet("Dimensions", pick(row, "Dimensions")), mdBullet("Format", pick(row, "File Format")), mdBullet("Logo placement", pick(row, "Logo Placement")), "");
  }
  if (mockups.length > 0) {
    lines.push("## Mockup Contexts", "");
    for (const row of mockups) lines.push(`### ${pick(row, "Context Name")}`, mdBullet("Scene", pick(row, "Scene Description")), mdBullet("Lighting", pick(row, "Lighting")), mdBullet("Environment", pick(row, "Environment")), mdBullet("Props", pick(row, "Props")), "");
  }
  lines.push("## Handoff Rules", "", "- Include logo variants, palette, typography, and usage rules.", "- Specify dimensions, file formats, material/finish assumptions, and production constraints.", "- Separate digital assets, print assets, source files, and exports.", "- Keep naming and versioning stable for approvals.");
  process.stdout.write(`${lines.join("\n")}\n`);
  return 0;
}

if (import.meta.url === pathToFileURL(process.argv[1]).href) process.exitCode = main();
